### PR TITLE
Feature/view controller lifecycle behaviors

### PR DESCRIPTION
**Pull request by ateliercw**

### CODE REVIEW

File: Pod/Classes/Lifecycle/DefaultBehaviors.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            }
+            DefaultBehaviors.injectBehaviors(behaviors, into: receiver, ignoring: ignored)
+        }
+
+        let swizzledIMP = imp_implementationWithBlock(unsafeBitCast(swizzledIMPBlock, to: AnyObject.self))
+        originalIMP = method_setImplementation(instanceViewDidLoad, swizzledIMP)
+
+    }
+
+    private static func injectBehaviors(_ behaviors: [ViewControllerLifecycleBehavior],
```
Comment (KingOfBrian) [inline Pod/Classes/Lifecycle/DefaultBehaviors.swift]: I like `inject(behaviors:` #pp, curious on when you decide to go one way or the other. Never clear to me!

File: Pod/Classes/Lifecycle/DefaultBehaviors.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                                        into viewController: UIViewController,
+                                        ignoring ignoredClasses: [UIViewController.Type]) {
+        for type in ignoredClasses + forcedIgnoredClasses {
+            guard !viewController.isKind(of: type) else {
+                return
+            }
+        }
+        // Prevents swizzing view controllers that are not subclassed from UIKit
+        let uiKitBundle = Bundle.init(for: UIViewController.self)
```
Comment (KingOfBrian) [inline Pod/Classes/Lifecycle/DefaultBehaviors.swift]: Why the `.init`? I like implicit init #pp

Comment (ateliercw) [inline Pod/Classes/Lifecycle/DefaultBehaviors.swift]: I start typing init to get the list of initializers from autocomplete, and then it leaves the init in from the suggestion

File: Pod/Classes/Lifecycle/DefaultBehaviors.swift (hunk 1 above)
Comment (ateliercw) [inline Pod/Classes/Lifecycle/DefaultBehaviors.swift]: force of habit

File: Pod/Classes/Lifecycle/DefaultBehaviors.swift [hunk 3]
```diff
@@ -0,0 +1,72 @@
+//
+//  DefaultBehaviors.swift
+//  Pods
+//
+//  Created by Michael Skiba on 2/23/17.
+//
```
Comment (ZevEisenberg) [inline Pod/Classes/Lifecycle/DefaultBehaviors.swift]: #? copyright?

**Pull request merged**